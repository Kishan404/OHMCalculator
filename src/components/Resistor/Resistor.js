import React, { Component } from 'react';
import PropTypes from 'prop-types';

const Resistor = (props) => {
    return (
        <div className="resistor">
            <span className="wire-start"></span>
            <span className="resitor-start"></span>
            <span className="resitor-middle"></span>
            <span className="resitor-end"></span>
            <span className="wire-end"></span>
            <span className={`resistor-color resistor-colorA ${props.colorinputs[0].toLowerCase()}`}></span>
            <span className={`resistor-color resistor-colorB ${props.colorinputs[1].toLowerCase()}`}></span>
            <span className={`resistor-color resistor-colorC ${props.colorinputs[2].toLowerCase()}`}></span>
            <span className={`resistor-color resistor-colorD ${props.colorinputs[3].toLowerCase()}`}></span>
        </div>
    );
}

Resistor.prototype = {
    colorinputs: PropTypes.array
}
export default Resistor;