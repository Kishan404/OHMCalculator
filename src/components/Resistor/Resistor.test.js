import React, { Component } from 'react';
const ReactTestRenderer = require('react-test-renderer');
import Resistor from './Resistor';

describe('Resistor result Snapshot test', () => {
    it('Should compare the component with a snapshot', () => {
      
    // const component = '<div class="resistor"><span class="wire-start"></span><span class="resitor-start"></span><span class="resitor-middle"></span><span class="resitor-end"></span><span class="wire-end"></span><span class="resistor-color resistor-colorA red"></span><span class="resistor-color resistor-colorB yellow"></span><span class="resistor-color resistor-colorC gray"></span><span class="resistor-color resistor-colorD none"></span></div>'
    // expect(component).toMatchSnapshot();
      
      const component = ReactTestRenderer.create(
        <Resistor 
            colorinputs = {['Red','Yellow','Gray','None']}
        />          
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });

