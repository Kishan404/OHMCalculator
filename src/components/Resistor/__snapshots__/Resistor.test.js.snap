// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resistor result Snapshot test Should compare the component with a snapshot 1`] = `
<div
  className="resistor"
>
  <span
    className="wire-start"
  />
  <span
    className="resitor-start"
  />
  <span
    className="resitor-middle"
  />
  <span
    className="resitor-end"
  />
  <span
    className="wire-end"
  />
  <span
    className="resistor-color resistor-colorA red"
  />
  <span
    className="resistor-color resistor-colorB yellow"
  />
  <span
    className="resistor-color resistor-colorC gray"
  />
  <span
    className="resistor-color resistor-colorD none"
  />
</div>
`;
